src/test/spec/unified_runner/matcher.rs

```diff
@@ -0,0 +1,197 @@
+use crate::bson::{doc, spec::ElementType, Bson};
+
+pub fn results_match(actual: Option<&Bson>, expected: &Bson) -> bool {
+    match expected {
+        Bson::Document(expected_doc) => {
+            if let Some((key, value)) = expected_doc.iter().next() {
+                if key.starts_with("$$") {
+                    return special_operator_matches((key, value), actual);
+                }
+            }
+            let actual = match actual {
+                Some(actual) => actual,
```

Comment:
maybe we can change this case to `Some(Bson::Document(actual))` and the other case to `_`. This let's us avoid the subsequent unwrap / panic.